app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -77,7 +77,12 @@ public void onClick(final View view) {
         }
 
         final ListView lv = (ListView) findViewById(R.id.listView);
-        lv.setAdapter(new ContactListAdapter(this, this.contacts));
+        try {
+            fillDatabaseWithDemoData();
+            lv.setAdapter(new ContactListAdapter(this, serviceLocator.getDatabase().getAllContacts()));
+        } catch (final IOException e) {
+            LOGGER.error("Failed to retrieve contacts from database", e);
```

Comment:
Filling the db could also go wrong,I don't mind this log but a more ambiguous one might be better.